pandas/core/base.py

```diff
@@ -894,12 +906,36 @@ def to_numpy(self):
         >>> ser = pd.Series(pd.Categorical(['a', 'b', 'a']))
         >>> ser.to_numpy()
         array(['a', 'b', 'a'], dtype=object)
+
+        Specify the `dtype` to control how datetime-aware data is represented.
+        Use ``dtype=object`` to return an ndarray of pandas :class:`Timestamp`
+        objects, each with the correct ``tz``.
+
+        >>> ser = pd.Series(pd.date_range('2000', periods=2, tz="CET"))
+        >>> ser.to_numpy(dtype=object)
+        array([Timestamp('2000-01-01 00:00:00+0100', tz='CET', freq='D'),
+               Timestamp('2000-01-02 00:00:00+0100', tz='CET', freq='D')],
+              dtype=object)
+
+        Or ``dtype='datetime64[ns]'`` to return an ndarray of native
+        datetime64 values. The values are converted to UTC and the timezone
+        info is dropped.
+
+        >>> ser.to_numpy(dtype="datetime64[ns]")
+        ... # doctest: +ELLIPSIS
+        array(['1999-12-31T23:00:00.000000000', '2000-01-01T23:00:00...'],
+              dtype='datetime64[ns]')
         """
         if (is_extension_array_dtype(self.dtype) or
                 is_datetime64tz_dtype(self.dtype)):
             # TODO(DatetimeArray): remove the second clause.
-            return np.asarray(self._values)
-        return self._values
+            result = np.asarray(self._values, dtype=dtype)
```

Comment:
technically this could cause a copy, right? so should we set copy=False if ``dtype`` is not None?

Reply:
Correct, we may be double copying. I'm not quite sure how to detect that. I'm not sure how to best handle this.

One option I considered is adding a method to the EA interface `_to_numpy` that returns `(ndarray, did_copy)`. Then this could be written as

```python
if is_extension_array_dtype(self.dtype):
    result, copied = self.array._to_numpy(dtype, copy)

if copy and not copied:
    return result
```

Do you think avoiding the double-copy is worth that complexity? Or perhaps there's a cleaner way I haven't thought of.